tools/topology/platform/intel/intel-generic-dmic-kwd.m4

```diff
@@ -0,0 +1,121 @@
+#
+# Topology for digital microphones array
+#
+
+include(`platform/intel/dmic.m4')
+
+# variable that need to be defined in upper m4
+# define(CHANNELS, `4')
+
+# define(DMIC_PCM_48k_ID, `99')
+# define(DMIC_PIPELINE_48k_ID, `4')
+# define(DMIC_DAI_LINK_48k_ID, `1')
+
+# define(DMIC_PCM_16k_ID, `7')
+# define(DMIC_PIPELINE_16k_ID, `9')
+# define(DMIC_PIPELINE_KWD_ID, `10')
+# define(DMIC_DAI_LINK_16k_ID, `2')
+
```

Comment:
This is a kind of framework code..which can be used across platforms.
The dailink definitions can vary per platform and has to align with the id's in machine driver.Is it a good idea to have these definitions as a part of m4 itself?
statically assigning the id's in a generic piece of code, forces the developer to use the same link id's in machine driver and this is not explicit.

Reply:
This is just some inline comments and example. Please note the `#`